package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,729 @@
+        ----------
+        step : int
+            Step number to calculate closest trajectory frame for.
+        ts : :class:`~MDAnalysis.coordinates.base.Timestep` object
+            Timestep from the trajectory the calculated frame number is to 
+            correspond to.
+
+        Returns
+        -------
+        int or None
+            Number of the trajectory frame closest (in time) to the given
+            auxiliary step. If the step index is out of range for the auxiliary
+            data, ``None`` is returned instead.
+
+        Notes
+        -----
+        Note that the returned frame number may be out of range for the 
+        trajectory.
+        """
+        if step not in range(self.n_steps):
+            return None 
+        offset = ts.data.get('time_offset', 0)
+        return int(math.floor((self.step_to_time(step)-offset+ts.dt/2.)/ts.dt))
+
+    def move_to_ts(self, ts):
+        """ Position auxiliary reader just before trajectory timestep *ts*.
+
+        Calling ``next()`` should read the first auxiliary step assigned to
+        (closest to) the trajectory timestep *ts* or, if no auxiliary steps are 
+        assigned to that timestep (as in the case of less frequent auxiliary
+        data), the first auxiliary step after *ts*.
+
+        Parameters
+        ----------
+        ts : :class:`~MDAnalysis.coordinates.base.Timestep` object
+            The trajectory timestep before which the auxiliary reader is to
+            be positioned.
+        """
+        # figure out what step we want to end up at
+        if self.constant_dt:
+            # if dt constant, calculate from dt/offset/etc
+            step = int(math.floor((ts.time-ts.dt/2-self.initial_time)/self.dt))
+            # if we're out of range of the number of steps, reset back
+            step = max(min(step, self.n_steps-1), -1)
+        else:
+            # otherwise, go through steps till we find the right one
+            for i in range(self.n_steps+1):
+                if self.step_to_frame(i) >= ts.frame:
+                    break
+            # we want the step before this
+            step = i-1
+        if step == -1:
+            self._restart()
+        else:
+            self._go_to_step(step)
+
+    def __getitem__(self, i):
+        """ Return the AuxStep corresponding to the *i*-th auxiliary step(s)
+        (0-based).
+        
+        *i* may be an integer (in which case the corresponding step is 
+        returned) or a slice (in which case an iterator is returned). """
+        if isinstance(i, int):
+            return self._go_to_step(i)
+        elif isinstance(i, (list, np.ndarray)):
+            return self._list_iter(i)
+        elif isinstance(i, slice):
+            return self._slice_iter(i)
+
+    def _list_iter(self, i):
+        for j in i:
+            yield self._go_to_step(j)
+
+    def _slice_iter(self, i):
+        #TODO - check slice is valid etc
+        for j in range(i.start, i.stop, i.step):
```

Comment:
I checked if the range function is properly imported at the top.  Please use `from six.moves import range` at the top of you file to override the default python 2 range to a version that returns a generator instead of returning a list.